torch/csrc/jit/passes/utils/check_alias_annotation.cpp

```diff
@@ -25,12 +25,22 @@ IValue deepCopy(const IValue& self) {
 
   // Lists of ivalues should recursively deep copy their contents
   if (self.isGenericList()) {
-    auto newList = c10::impl::GenericList(c10::impl::deprecatedUntypedList());
-    newList.reserve(self.toGenericListRef().size());
-    for (const IValue& value : self.toGenericListRef()) {
-      newList.push_back(deepCopy(value));
+    auto source = std::move(self).toGenericList();
+    auto deepCopyGenericList = [&] (c10::impl::GenericList& dest) {
+      dest.reserve(source.size());
+      for (const IValue& value : source) {
+        dest.push_back(deepCopy(value));
+      }
+    };
+    if (source._elementType().has_value()) {
+      auto newList = c10::impl::GenericList(*source._elementType());
```

Comment:
can it be done without lambda?

```
auto newList = source._elementType().has_value() ? c10::impl::GenericList(*source._elementType()) : c10::impl::GenericList(c10::impl::deprecatedUntypedList());
...
```

Reply:
I hadn't thought of the conditional expression. Thanks, will use that instead.